### PR TITLE
Created ArrayList class which implements the List interface.

### PR DESCRIPTION
**Pull request by Alex-Solod**

### CODE REVIEW

**Review by Ivan95kos — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,48 +1,107 @@
...
     public void set(T value, int index) {
-
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
```
[the remark this answers is not in the document]
Comment (Alex-Solod) [reply src/main/java/core/basesyntax/ArrayList.java:62]: Can u explain what is <checkIndex(index);> and <checkIndexEqualsSize(index);> ?

[the remark this answers is not in the document]
Comment (Ivan95kos) [reply src/main/java/core/basesyntax/ArrayList.java:62]: These are private methods to check the index

```
    private void checkIndex(int index) {
        if (index > size || index < 0) {
            throw new ArrayListIndexOutOfBoundsException("Index is wrong: " + index);
        }
    }

    private void checkIndexEqualsSize(int index) {
        if (index == size) {
            throw new ArrayListIndexOutOfBoundsException("Index is wrong: " + index);
        }
    }
```

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double SIZE_GROW = 1.5;
+    private Object[] elements;
```
[the remark this answers is not in the document]
Comment (Ivan95kos) [reply src/main/java/core/basesyntax/ArrayList.java:9]: not fixed, you must use parameterized type `T`

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -1,48 +1,107 @@
...
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
```
[the remark this answers is not in the document]
Comment (Ivan95kos) [reply src/main/java/core/basesyntax/ArrayList.java:44]: not fixed

File: src/main/java/core/basesyntax/ArrayList.java [hunk 4]
```diff
@@ -1,48 +1,105 @@
...
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/ArrayList.java:44]: You use this check several times
```suggestion
        checkIndex(index);
```

File: src/main/java/core/basesyntax/ArrayList.java [hunk 5]
```diff
@@ -1,48 +1,107 @@
...
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
```
[the remark this answers is not in the document]
Comment (Ivan95kos) [reply src/main/java/core/basesyntax/ArrayList.java:26]: not fixed

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
please, fix previous comments

**Pull request closed without merging**